internal/shutdown/signals.go

```diff
@@ -0,0 +1,15 @@
+//go:build unix
+
+package shutdown
+
+import (
+	"os"
+	"syscall"
+)
+
+// GetSignals returns the appropriate signals to catch for a clean shutdown, dependent on the OS.
+//
+// On Unix-like operating systems, it is important to catch SIGTERM in addition to SIGINT.
+func GetSignals() []os.Signal {
+	return []os.Signal{os.Interrupt, syscall.SIGTERM}
```

Comment:
Should this be https://pkg.go.dev/syscall#SIGINT? Should we add SIGKILL also?

So, far athens does not hav ea lot of linux specific code afaik? Does it make sense to find out the os on runtime rather than have build tags and have a single package.

Reply:
`os.Interrupt` is mapped to `syscall.SIGINT`. Also, containers will get a `syscall.SIGTERM` to indicate the container should shutdown, so in general it should be handled along side `os.Interrupt`.  I don't know that `syscall.SIGTERM` exists in all OSes, so it is safer to use build tags IMO. If it can be proven that it always exists, then the build tags would not be needed.

`syscall.SIGKILL` can in general not be caught, so there is not much point in trying.

Reply:
I was suggesting something like this: https://github.com/gomods/athens/blob/a1553999bb8bfc57d9962ca3ee3ed992640b8f2b/pkg/module/prepare_env.go#L37

Reply:
The issue I see there, is that should the signal not exist in the `syscall` package for the OS, you would not be able to build on that platform anymore, where build tags would just exclude the code from the build process in non-supporting OSes.

Reply:
To be more clear, `SIGTERM` in the `syscall` package (and all signals in the syscall package) is protected by platform specific build tags, and so there is no guarantee that it exists for all OSes.

Reply:
Right, that is why I have it protected by build tags. This is what I had to do to get it shutting down properly in k8s.

Reply:
Thanks! Lgtm